### PR TITLE
[Fixes #206] Clone Project

### LINKED ISSUE #206: Clone Project option in home page
## Feature Request

Is your feature request related to an existing [Issue](https://www.github.com/Auto-DL/Generator/issues) or a new implementation?

### Feature Description

Add another option "Clone" in the image given below and on click maybe open a model which has 3 checkboxes ( 
Model Layers
Preprocessing params
Hyperparams
)
which clones the respective part of the project and if all are selected clones the whole project.
image

### PR DESCRIPTION
**Pull request by McTechie**
## Pull Request

fixes #206 

### Description
Project Clone option added to Actions. The cloned project may have a new **name**, **output file name**, and **project description** along with capabilities for cloning the **Model Layers**, **Preprocessing Params** and **Hyperparams** as per the owner's choice (using a Checklist).

### Cloning UI
Clone Option
Clone Modal Step 1
Clone Modal Step 2
Clone Completion

### Checklist:
- [x] Followed the guidelines mentioned in our [Contributing document](../CONTRIBUTING.md)
- [x] Ensured that there aren't other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change
- [x] This PR is linked to an [Issue](https://www.github.com/Auto-DL/Generator/issues) and the issue number has been mentioned in the title and body of the PR
- [x] The code is clean and docstrings have been added or updated as required
- [x] Code is linted/formatted locally prior to submission (using `prettier` and `black`)

### CODE REVIEW

File: BackEndApp/v1/views.py [hunk 1]
```diff
@@ -299,6 +299,79 @@ def create_project(request):
...
 
 
+@api_view(["POST"])
+@is_authenticated
+def clone_project(request):
+    try:
+        username = request.data.get("username")
+        modelLayers = request.data.get("modelLayers")
+        preprocessingParameters = request.data.get("preprocessingParameters")
+        hyperParameters = request.data.get("hyperParameters")
```
Comment (ADI10HERO) [inline BackEndApp/v1/views.py:309]: Please use snake case variable names 
Eg. `model_layers`

**Review by ADI10HERO — COMMENTED** [review]
A minor change :) Rest everything looks good as per me. Requesting @rajraj889  to review as well

File: BackEndApp/v1/views.py (hunk 1 above)
Comment (McTechie) [reply BackEndApp/v1/views.py]: Yes, I have appropriately changed the case wherever required :D

**Review by ADI10HERO — APPROVED** [review]

**Review by rajraj889 — APPROVED** [review]
image
Validation needed for this part.

image
Atleast 1 checkbox must be checked for a clone to be created.

Make an issue for the respective.

**Pull request merged**